src/sparseml/pytorch/optim/modifier_pruning.py

```diff
@@ -1762,3 +1763,167 @@ def _should_log(
             self._last_logged_epoch != math.floor(epoch)
             or self._last_logged_layers_replaced != self._layers_replaced
         )
+
+
+@PyTorchModifierYAML()
+class ACDCPruningModifier(GMPruningModifier):
+    """
+    Implementation of
+    Alternating Compressed/DeCompressed Training of Deep Neural Networks:
+    https://arxiv.org/pdf/2106.12379.pdf.
+    AC/DC performs co-training of sparse and dense models, and can return both an
+    accurate sparse model, and a dense model.
+    | Sample yaml:
+    |   !ACDCPruningModifier
+    |       compression_sparsity: 0.95
+    |       start_epoch: 0
+    |       end_epoch: 100
+    |       update_frequency: 5
+    |       params: __ALL_PRUNABLE__
+
+    :param compression_sparsity: The sparsity enforced during the compression phase.
+    :param start_epoch: The epoch to start the modifier at
+    :param end_epoch: The epoch to end the modifier at
+    :param update_frequency: The length (in epochs) of compression/decompression phase.
+    :param params: A list of full parameter names or regex patterns of names to apply
+        pruning to.  Regex patterns must be specified with the prefix 're:'. __ALL__
+        will match to all parameters. __ALL_PRUNABLE__ will match to all ConvNd
+        and Linear layers' weights. If a sparsity to param mapping is defined by
+        final_sparsity, then params should be set to []
+    """
+
+    def __init__(
+        self,
+        # because method does not involve any interpolation
+        # compression sparsity is a single float.
```

Comment:
nit: let's move this comment to line 1806

Reply:
Agree, makes sense.